src/GovUk.Education.ExploreEducationStatistics.Data.Services/FilterAndIndicatorViewModelBuilders.cs

```diff
@@ -0,0 +1,296 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Data.Model;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels.Meta;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Services;
+
+public class FilterAndIndicatorViewModelBuilders
+{
+    public static class FiltersViewModelBuilder
+    {
+        public static Dictionary<string, FilterMetaViewModel> BuildFiltersFromFilterItems(
+            IEnumerable<FilterItem> values,
+            IEnumerable<FilterSequenceEntry>? sequence = null)
+        {
+            var filters = GroupFilterItemsByFilter(values);
+            return BuildFilters(filters, sequence);
+        }
+
+        public static Dictionary<string, FilterMetaViewModel> BuildFilters(IEnumerable<Filter> values,
+            IEnumerable<FilterSequenceEntry>? sequence = null)
+        {
+            FilterMetaViewModel BuildViewModel(Ordered<Filter, FilterSequenceEntry> input, int index)
```

Comment:
Not a biggy, but how come we haven't inlined this into a lambda on `resultSelector`? It's just a little unusual to see a full function declaration like this?

Seems like we've also done this in the other methods as well, so same question applies for those too